packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_set_view.dart

```diff
@@ -2,17 +2,23 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+
 import 'package:flutter/material.dart';
-import 'package:vm_service/vm_service.dart';
 
 import '../../../../shared/analytics/constants.dart';
 import '../../../../shared/common_widgets.dart';
 import '../../../../shared/primitives/utils.dart';
 
-typedef SampleObtainer = InstanceRef Function();
+abstract class ClassSampler {
```

Comment:
why do we need an abstract class for this? will there be another class that extends ClassSampler at some point?

Reply:
Yes, table that diffs snapshots, will need another implementation.